web/pdf_cursor_tools.js

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2017 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GrabToPan } from './grab_to_pan';
+
+const CursorTool = {
+  SELECT: 0, // The default value.
+  HAND: 1,
+  ZOOM: 2,
+};
+
+/**
+ * @typedef {Object} PDFCursorToolsOptions
+ * @property {HTMLDivElement} container - The document container.
+ * @property {EventBus} eventBus - The application event bus.
+ * @property {BasePreferences} preferences - Object for reading/writing
+ *                                           persistent settings.
+ */
+
+class PDFCursorTools {
+  /**
+   * @param {PDFCursorToolsOptions} options
+   */
+  constructor({ container, eventBus, preferences, }) {
+    this.container = container;
+    this.eventBus = eventBus;
+
+    this.active = CursorTool.SELECT;
+    this.activeBeforePresentationMode = null;
+
+    this.handTool = new GrabToPan({
+      element: this.container,
+    });
+
+    this._addEventListeners();
+
+    Promise.all([
+      preferences.get('cursorToolOnLoad'),
+      preferences.get('enableHandToolOnLoad')
+    ]).then(([cursorToolPref, handToolPref]) => {
+      // If the 'cursorToolOnLoad' preference has not been set to a non-default
+      // value, attempt to convert the old 'enableHandToolOnLoad' preference.
+      // TODO: Remove this conversion after a suitable number of release.
+      if (handToolPref === true) {
+        preferences.set('enableHandToolOnLoad', false);
+
+        if (cursorToolPref === CursorTool.SELECT) {
+          cursorToolPref = CursorTool.HAND;
+          preferences.set('cursorToolOnLoad', cursorToolPref).catch(() => { });
+        }
+      }
+      this.switchTool(cursorToolPref);
+    }).catch(function(reason) { });
```

Comment:
Nit: make this an arrow function as well since you do the same for `preferences.set` above. It may also be a regular function, as long as both are the same.